### ISSUE TITLE
Fix compilation issue with Windows / MSYS

### ISSUE THREAD

**Issue by cschneid** [scoutapp/scout_apm_ruby#111]
MSYS doesn't support the posix sys/resource.h header, and hence fails to compile.

We should have some sort of fallback if that header isn't available.

```
Temporarily enhancing PATH for MSYS/MINGW...
Building native extensions.  This could take a while...
ERROR:  Error installing scout_apm:
        ERROR: Failed to build gem native extension.
    current directory: C:/Ruby24-x64/lib/ruby/gems/2.4.0/gems/scout_apm-2.1.26/ext/allocations
C:/Ruby24-x64/bin/ruby.exe -r ./siteconf20170719-15944-1n0ro3h.rb extconf.rb
checking for ruby/ruby.h... yes
creating Makefile
current directory: C:/Ruby24-x64/lib/ruby/gems/2.4.0/gems/scout_apm-2.1.26/ext/allocations
make "DESTDIR=" clean
current directory: C:/Ruby24-x64/lib/ruby/gems/2.4.0/gems/scout_apm-2.1.26/ext/allocations
make "DESTDIR="
generating allocations-x64-mingw32.def
compiling allocations.c
allocations.c:13:10: fatal error: sys/resource.h: No such file or directory
 #include <sys/resource.h> // is this needed?
          ^~~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:242: allocations.o] Error 1
make failed, exit code 2
Gem files will remain installed in C:/Ruby24-x64/lib/ruby/gems/2.4.0/gems/scout_apm-2.1.26 for inspection.
Results logged to C:/Ruby24-x64/lib/ruby/gems/2.4.0/extensions/x64-mingw32/2.4.0/scout_apm-2.1.26/gem_make.out
```

**Comment by LiamReynaud** [comment]
Hi,
I have the same problem, trying to install fatal error: sys/resource.h: No such file or directory
with version 2.1.32 did you find a solution ?

**Comment by cschneid** [comment]
@LiamReynaud - we merged #92 into our 3.0.0.preX branch of the agent, but it wasn't backported to our 2.x branch. I just backported it, but haven't released a new version yet. Can you point your scout_apm gem at master in your Gemfile and try to install?

If this works for you, I'll cut a new release as `2.1.33`.

`gem 'scout_apm', git: "https://github.com/scoutapp/scout_apm_ruby.git" , branch: 'master'`

**Comment by alindeman** [comment]
I think this might have already been fixed, as I see the changes from #92 in master.

To confirm, I spun up an EC2 instance running Windows and installed the latest version of `scout_apm`:

![gem install scout_apm on windows](https://user-images.githubusercontent.com/395621/36825232-60f99238-1cd4-11e8-885e-0b7f24e74eec.png)

/cc: @dlanderson

**Comment by cschneid** [comment]
@alindeman - thank you for checking that. 👍
